src/quasi-newton.jl

```diff
@@ -132,6 +172,8 @@ end
 
 NLPModels.neval_hprod(nlp::LBFGSModel) = nlp.op.nprod
 NLPModels.neval_hprod(nlp::LSR1Model) = nlp.op.nprod
+NLPModels.neval_hprod(nlp::DiagonalQNModel) = nlp.op.nprod
+NLPModels.neval_hprod(nlp::SpectralGradientModel) = nlp.op.nprod
```

Comment:
Can you also add tests here https://github.com/JuliaSmoothOptimizers/NLPModelsModifiers.jl/blob/157da2ef8a28bb4ac5b6869de4425c32bb212333/test/nlp/quasi-newton.jl#L40 to improve coverage